doc/srv6/srv6_vpn.md

```diff
@@ -0,0 +1,284 @@
+
+The minimal FRR version for SRv6 VPN is 8.4, which you could get SRv6 VPN PoC functionalities. FRR version 8.5 or 9.0 would provide Alibaba deployment feature parity.
+
+## SAI
+
+Current SRv6 SAI doesn't have VPN support. SRv6 VPN SAI would be contributed by Cisco team. The SAI object model for SRv6 VPN is shown in the following diagram. The detail information about SRv6 VPN SAI APIs could be found in SAI's HLD document. ([https://github.com/rameshms-work/SAI/blob/rameshms/srv6\_vpn\_enhancements/doc/SAI-IPv6-Segment-Routing-VPN.md](https://github.com/rameshms-work/SAI/blob/rameshms/srv6_vpn_enhancements/doc/SAI-IPv6-Segment-Routing-VPN.md))
+
+![image](images/2SAI.png)
+
+## SWSS and SONiC related changes
+
+The main SRv6 SWSS design is documented via [SRv6 HLD](https://github.com/sonic-net/SONiC/blob/master/doc/srv6/srv6_hld.md). This HLD focuses on the changes needed to enable SRv6 VPN on the top of existing SRv6 changes.
+
+# SRv6Orch and RouteOrch related Changes
+
+The overall work flow diagram is shown in the diagram below.
+
+![image](images/3SWSS_workflow.png)
+
+## APP\_DB
+
+### New SRv6\_POLICY\_TABLE
+
+SRv6\_SID\_LIST\_TABLE and SRv6\_MY\_SID\_TABLE have already been supported. The detail information could be found in [SRv6 HLD](https://github.com/sonic-net/SONiC/blob/master/doc/srv6/srv6_hld.md). For SRv6 traffic steering, we add a new SRv6 POLICY table. 
+
+Description : new table to hold SRv6 policy
+
+Schema: 
+
+    ; New table
+    ; holds SRv6 policy
+    
+    key = SRV6_POLICY_TABLE|policy_name ;
+    segment = SRv6_SID_LIST.key,   ; List of segment names, separated by ','
+    weight  = weight_list          ; List of weights, separated by ','
+    
+
+### Modify ROUTE\_TABLE
+
+Description:  vpn\_sid and policy fields are added to existing route table schema.
+
+Schema:
+
+    ;Stores a list of routes
+    ;Status: Mandatory
+    
+    key = ROUTE_TABLE:VRF_NAME:prefix ;
+    nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+    intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+    vni_label = VRF.vni            ; zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
+    router_mac = mac_address       ; zero or more remote router MAC address separated by ',' (empty value for non-vxlan next-hops)
+    blackhole = BIT                ; Set to 1 if this route is a blackhole (or null0)
+    segment = SRV6_SID_LIST.key    ; New optional field. List of segment names, separated by ',' 
+    seg_src = address              ; New optional field. Source addrs for sid encap
+    vpn_sid = vpn_sid              ; New optional field to add vpn_sid to learnt BGP routes
+    policy = policy                ; New optional field to add policy name to learnt BGP routes
```

Comment:
as mentioned above, I am not sure if policy should really belong to orchagent.. There is no policy notion in orchagent for regular routing

Reply:
I quote @reshmaintel  's previous comments here "SRv6Orch will continue to do other tasks for MY_SID and SRv6 policy -processing," .  The current thought process is to continue with current design. We could discuss this topic in routing WG if you want. 

For 202205, we will upstream SRv6 VPN only. Policy will come in the future release.